github-comment-fetcher/src/main/java/com/tzachz/commentcounter/PullRequestCache.java

```diff
@@ -18,6 +18,7 @@
 public class PullRequestCache {
 
     public static final GHPullRequest UNKNOWN = new GHPullRequest(new GHUser("UNKNOWN", ""));
+    public static final GHPullRequest NOT_FOUND = new GHPullRequest(null);
```

Comment:
actually, I've grown accustomed to _never using nulls_ - as they hide the "optionality" of a value from the code reader (and the compiler!). I usually use Guava's `Optional` as the type for optional values, but since you've bumped this to Java 8 I guess Java's built in `Optional` would make more sense. Care to try and change `GHPullRequest` to expect an `Optional<GHUser>` argument instead of a `GHUser` to support this? Might incur a few more changes, but is eventually safer.

This probably applies to `GitHubApiFacade.getPullRequest` too - should return an `Optional<GHPullRequest>`.

What do you think?